### PR TITLE
Pull request for Members branch 

### PR DESCRIPTION
**Pull request by Datagirlcmr**

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class PostsController < ApplicationController
+  before_action :is_signed_in?, only: [:new, :create]
+
+  def new
+    @post = Post.new()
```
Comment (Redvanisation) [inline app/controllers/posts_controller.rb]: Remove the parenthesis it's `Post.new`

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  def create
+    post = Post.new(title: params[:post][:title], body: params[:post][:body], user_id: current_user.id)
```
Comment (Redvanisation) [inline app/controllers/posts_controller.rb]: You could improve this by creating the **private** `post_params` method, and changing this line after that to be
`@post = current_user.posts.build(post_params)`

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -1,2 +1,17 @@
...
+      if user && user.authenticated?(cookies[:remember_token])
+        sign_in user
+        @current_user = user
+      end
+    end
+  end
+
+  def signed_in?
+    !current_user.nil?
+  end
```
Comment (Redvanisation) [inline app/helpers/application_helper.rb]: There should be no application logic in the views, you already implemented those methods in the `ApplicationController`

**Review by Redvanisation — CHANGES_REQUESTED** [review]
# Changes required  ♻️ 

### Hello team  🌟

**Very nice work on this project, however, the following issues need to be fixed before moving on ⬇️** 

#### PostsController `create` action:

* [ ] it makes sure that the user_id for post is the one from the signed_in user ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_create_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)).

* [ ] You should also use the `post_params` private method.

#### UsersController`create` action

* [ ] it has a create action. Must use `user_params` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_up_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/))

* [ ] There should be NO application logic in the views nor helpers (no `Post.new`, `Comment.create`, `current_user.friends.find_by(...)`...). Everything connected with data processing should be encapsulated in models and used in controllers, check [this article](http://codefol.io/posts/Where-Do-I-Put-My-Code) for reference.

**[OPTIONAL]:** Update the routes and add a `root` route so the landing page would be the login view for example instead of the new rails app page.

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Congratulations :clap: 

You did it :smile: :+1: 
Good luck with the next step.

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Pull request merged**